server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -846,11 +904,31 @@ private CompletionStage<Result> renderErrorOrRedirectToRequestedPage(
                       : MessageKey.TOAST_MAY_QUALIFY.getKeyName(),
                   roApplicantProgramService.getProgramTitle()));
     }
+    return getRequestedPage(
+        profile,
+        applicantId,
+        programId,
+        blockId,
+        inReview,
+        applicantRequestedAction,
+        roApplicantProgramService,
+        flashingMap);
+  }
 
-    if (applicantRequestedAction == ApplicantRequestedAction.REVIEW_PAGE) {
+  /** Returns the correct page based on the given {@code applicantRequestedAction}. */
+  private CompletionStage<Result> getRequestedPage(
```

Comment:
**nitpick:** Should we name redirectToRequestedPage?

Reply:
+1

Reply:
Great name! Done.